framework/src/transfers/MultiAppFieldTransferInterface.C

```diff
@@ -0,0 +1,396 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "MultiAppFieldTransferInterface.h"
+#include "MooseTypes.h"
+#include "FEProblem.h"
+#include "MultiApp.h"
+#include "MooseMesh.h"
+#include "UserObject.h"
+#include "NearestPointIntegralVariablePostprocessor.h"
+
+template <>
+InputParameters
+validParams<MultiAppFieldTransferInterface>()
+{
+  InputParameters params = validParams<MultiAppTransfer>();
+  params.addRequiredParam<std::vector<AuxVariableName>>(
+      "variable", "The auxiliary variable to store the transferred values in.");
```

Comment:
This is a quite big thing. Should we allow the target variable to be a primal variable? I think we should.